Source/Core/Core/Debugger/BranchWatch.h

```diff
@@ -119,7 +122,12 @@ class BranchWatch final  // Class is final to enforce the safety of GetOffsetOfR
   using SelectionInspection = BranchWatchSelectionInspection;
 
   bool GetRecordingActive() const { return m_recording_active; }
-  void SetRecordingActive(const CPUThreadGuard& guard, bool active) { m_recording_active = active; }
+  void SetRecordingActive(const CPUThreadGuard& guard, bool active)
+  {
+    auto& system = guard.GetSystem();
+    system.GetJitInterface().ClearCache(Core::CPUThreadGuard{system});
```

Comment:
You already have a CPUThreadGuard. Please use that one instead of constructing a new one.